packages/xrpl/src/sugar/submit.ts

```diff
@@ -248,8 +182,42 @@ function isSigned(transaction: Transaction | string): boolean {
   return tx.SigningPubKey != null && tx.TxnSignature != null
 }
 
-// initializes a transaction for a submit request
-async function getSignedTx(
+/**
+ * Retrieves the signed transaction from the client.
+ *
+ * @param client - The client from which to retrieve the signed transaction.
+ * @param transaction - The transaction to retrieve. It can be either a Transaction object or
+ * a string (encode from ripple-binary-codec) representation of the transaction.
+ * @param [options={}] - Optional. Additional options for retrieving the signed transaction.
+ * @param [options.autofill=true] - Optional. Determines whether the transaction should be autofilled (true)
+ * or not (false). Default is true.
+ * @param [options.failHard] - Optional. If true, and the transaction fails locally, do not retry or relay the
```

Comment:
Potentially tangential to this PR, but it seems that `options.failHard` is never used and doesn't really make sense in this function (other than allowing opts to be passed in directly from the higher level function) - This function doesn't submit to ledger, so it can't fail like that.